src/storage/src/source/kafka.rs

```diff
@@ -541,29 +555,30 @@ impl KafkaSourceReader {
     /// fair manner.
     fn poll_from_next_queue(
         &mut self,
-    ) -> Result<Option<SourceMessage<Option<Vec<u8>>, Option<Vec<u8>>, ()>>, anyhow::Error> {
+    ) -> Result<
```

Comment:
why do we prefer a nested `Result` here?   Is there something that prevents flattening?

Reply:
For http://sled.rs/errors.html -shaped reasons, basically: the outer errors are the sort we want to percolate up and use `?` for, and the inner errors should be handled right away from the caller. Collapsing to the same variant would lose the distinction, and coming up with a custom `enum` to distinguish seemed if anything more error prone. (But is the best alternative IMO if we don't like the nested-result approach!)

Reply:
makes sense!